lib/dry/system/plugins/dependency_graph.rb

```diff
@@ -0,0 +1,39 @@
+require 'dry/system/constants'
+require 'dry/system/plugins/dependency_graph/strategies'
+
+module Dry
+  module System
+    module Plugins
+      # @api public
+      module DependencyGraph
+        # @api private
+        def self.extended(system)
+          super
+
+          system.use(:notifications)
+
+          system.after(:configure) do
+            self[:notifications].register_event(:resolved_dependency)
+            self[:notifications].register_event(:registered_dependency)
+
+            system.strategies(Stratagies.with_notifications(system[:notifications]))
+          end
+        end
+
+        # @api private
+        def self.dependencies
+          'dry/events/publisher'
+        end
+
+        def register(key, contents = nil, options = {}, &block)
+          unless key.to_sym == :notifications
```

Comment:
This should be configurable as I'm pretty sure we'd like to skip it for other deps.

Reply:
good point. will add it 👍